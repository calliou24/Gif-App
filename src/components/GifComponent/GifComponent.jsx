import React from "react";
import { PropTypes } from "prop-types";

import styles from './gif-component.module.css'

export const GifComponent = ({ title, url }) => {
  return (
    <li className={styles.item}
      onClick={() => {
        navigator.clipboard.writeText(url);
      }}
    >
        <img src={url} alt={title}></img>
    </li>
  );
};

GifComponent.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
