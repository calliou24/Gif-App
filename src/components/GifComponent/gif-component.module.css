.item {
  position: relative;
  cursor: pointer;
}

.item::after {
  content: "Copy URL";
  position: absolute;
  background-color: #0006;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  opacity: 0;
  transition: all .2s;
}

.item:hover::after {
  opacity: 1;

}
