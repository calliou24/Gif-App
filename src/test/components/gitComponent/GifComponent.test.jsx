import React from 'react'
import { shallow } from 'enzyme'
import { beforeEach, describe, expect, test, vi } from 'vitest'

import { GifComponent } from './../../../components/GifComponent/GifComponent'

describe('Button component', () => {
    

    let title = 'TITLE'
    let url = 'https://music.youtube.com'
    const wrapper = shallow(<GifComponent title={title} url={url}/>)

    test('Snapshot', ()=> {
        expect(wrapper).toMatchSnapshot()
    })

    test('My title should match width the GifComponent "p" contetnt', ()=> {

        const wrapperText = wrapper.find('p').text()
        expect(wrapperText).toBe(title)
    })
    test('My image src should be the same as url',()=> {
        
        const src = wrapper.find('img').prop('src')
        // console.log(src)
        // console.log(url)
        expect(src).toBe(url)
    })
    test('should return the className "animate__bounceInLeft"', () => {
        
        const wrapperClassName = wrapper.find("li").prop('className')
        
        let classNameTest = 'animate__bounceInLeft'
        //expect(wrapperClassName.className).includes(classNameTest) This could be a way to test the className prop
        expect(wrapperClassName.includes(classNameTest)).toBe(true) //this one is another but i like the first one
     })

})
