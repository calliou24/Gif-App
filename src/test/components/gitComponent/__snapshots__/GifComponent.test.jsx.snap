// Vitest Snapshot v1

exports[`Button component > Snapshot 1`] = `
<li
  className="animate__animated animate__bounceInLeft"
  onClick={[Function]}
>
  <div
    className="image-cont"
  >
    <img
      alt="TITLE"
      src="https://music.youtube.com"
    />
  </div>
  <div>
    <p>
      TITLE
    </p>
  </div>
</li>
`;
